### PR TITLE
API docs landing page

### LINKED ISSUE #1580: API docs landing page
Today, we list each package in the TOC under top-level Node.js and Python items, which really bifurcates the APIs between Node.js vs Python:

Screen Shot 2019-08-19 at 9 14 27 AM

Instead, we'll stop showing these packages in the TOC (#831) and add a single landing page for the API Reference, with lang chooser at the top of the page. This landing page will better present our various packages, e.g. under a AWS heading it will list, aws, awsx, and eks packages.

### PR DESCRIPTION
**Pull request by justinvp**
This change removes the API packages from the TOC and instead lists them on a single API Reference landing page, which has a language chooser to see the TS/JS vs. Python packages.

This largely just combines the existing two pages into a single combined page w/ lang chooser and eliminates the items from the TOC.

It could definitely be spruced up further, e.g.:
 - Adding logos for the "tier 1" clouds, possibly showing them like the Tutorials landing page, or having more of a separation between the mainline clouds and other providers
 - Listing the packages in 2 or 3 columns with nicer package-specific purple "bullet" like the bullets for APIs in #1669
 - Tighter copy at the top of the page

Fixes #1580

### CODE REVIEW

**Comment by justinvp** [comment]
@bermudezmt, @lukehoban, is this along the lines of what you were thinking, or did you have something else in mind?

File: content/docs/reference/pkg/_index.md [hunk 1]
```diff
@@ -11,18 +12,94 @@ as higher-level APIs that make it easier to deliver cloud applications and
...
+* [**Packet** (`pulumi_packet`)]({{< relref "python/pulumi_packet" >}})
+* [**PostgreSQL** (`pulumi_postgresql`)]({{< relref "python/pulumi_postgresql" >}})
+* [**TLS** (`pulumi_tls`)]({{< relref "python/pulumi_tls" >}})
+* [**vSphere** (`pulumi_vsphere`)]({{< relref "python/pulumi_vsphere" >}})
+{{% /lang %}}
+
+### Cloud-Agnostic Packages
+
+{{% lang nodejs %}}
+Pulumi offers a highly productive, cloud-agnostic package for container and serverless oriented programming in the
```
Comment (bermudezmt) [inline content/docs/reference/pkg/_index.md]:
```suggestion
Pulumi offers a highly productive, cloud-agnostic package for container and serverless programming in the
```

File: content/docs/reference/pkg/_index.md [hunk 2]
```diff
@@ -11,18 +12,94 @@ as higher-level APIs that make it easier to deliver cloud applications and
 infrastructure.
 
 These APIs are available as packages in your chosen language's package manager
```
Comment (bermudezmt) [inline content/docs/reference/pkg/_index.md]:
```suggestion
These APIs are available as packages in your chosen language's package manager
```

File: content/docs/reference/pkg/_index.md [hunk 3]
```diff
@@ -11,18 +12,94 @@ as higher-level APIs that make it easier to deliver cloud applications and
...
-&mdash; NPM for Node.js and PyPI for Python. There is a dedicated package for
+&mdash; npm for TypeScript/JavaScript and PyPI for Python. There is a dedicated package for
```
Comment (bermudezmt) [inline content/docs/reference/pkg/_index.md]:
```suggestion
---npm for TypeScript/JavaScript and PyPI for Python. There is a dedicated package for
```

File: content/docs/reference/pkg/_index.md [hunk 4]
```diff
@@ -11,18 +12,94 @@ as higher-level APIs that make it easier to deliver cloud applications and
...
-![Pulumi Library Architecture](/images/docs/reference/pkg-arch-layers.png)
+{{< langchoose nogo >}}
 
-You can read further documentation specific to each language here:
+### General Purpose Packages
 
-* [JavaScript/TypeScript](nodejs)
-* [Python](python)
-* Go (coming soon!)
+The Pulumi SDK package is used for accessing the core programming model around resources, configuration, etc.
```
Comment (bermudezmt) [inline content/docs/reference/pkg/_index.md]: I don't like using etc. in technical documentation, so how about "configuration, and other components"?

Comment (justinvp) [inline content/docs/reference/pkg/_index.md]:
```suggestion
The Pulumi SDK package is used for accessing the core programming model around resources, configuration, and other components
```

**Pull request merged**